### PR TITLE
CLID-85: Change string replacement in registry config to use templates

### PR DESCRIPTION
**Pull request by lmzuccarelli**
# Description

Update to change string replacement with text/template

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Standard unit tests

## Expected Outcome

TestExecutorSetupLocalStorage - should pass

### CODE REVIEW

File: v2/pkg/cli/executor.go [hunk 1]
```diff
@@ -450,27 +451,30 @@ http:
       #path: /etc/registry
 `
 
-	if _, err := os.Stat(o.LocalStorageDisk); err != nil {
-		// something went wrong
-		return fmt.Errorf("error using the local storage folder for caching")
-	}
```
Comment (sherine-k) [inline v2/pkg/cli/executor.go:456]: this can be saved. it has value

File: v2/pkg/cli/executor.go [hunk 2]
```diff
@@ -450,27 +451,30 @@ http:
...
-	configYamlV01 = strings.Replace(configYamlV01, "$$PLACEHOLDER_ROOT$$", o.LocalStorageDisk, 1)
-	configYamlV01 = strings.Replace(configYamlV01, "$$PLACEHOLDER_PORT$$", strconv.Itoa(int(o.Opts.Global.Port)), 1)
-	if o.Opts.Global.LogLevel != "trace" {
-		configYamlV01 = strings.Replace(configYamlV01, "$$PLACEHOLDER_LOG_LEVEL$$", o.Opts.Global.LogLevel, 1)
-	} else {
-		configYamlV01 = strings.Replace(configYamlV01, "$$PLACEHOLDER_LOG_LEVEL$$", "info", 1)
```
Comment (sherine-k) [inline v2/pkg/cli/executor.go:462]: trace is not a log level understood , so you need some sort of translation for it. that is why, we cannot simply use: 
```
rc := RegistryConfig{
		LocalStorageDisk: o.LocalStorageDisk,
		LocalStoragePort: int(o.Opts.Global.Port),
		LogLevel:         o.Opts.Global.LogLevel,
		LogAccessOff:     o.Opts.Global.LogLevel != "debug",
	}
```

**Review by sherine-k — COMMENTED** [review]
Hi @lmzuccarelli 
This is a much nicer approach!
 Beware some pitfalls that I highlighted here.

**Review by aguidirh — APPROVED** [review]
Thanks for this change @lmzuccarelli :tada: 

Using templates here was a wise choice.

**Pull request merged**